### PR TITLE
presentationBranch

### PR DESCRIPTION
**Pull request by mustiwebp**
Changes/features:
- myMedication screen
- family members screen
- physicians screen
- overview screen

### CODE REVIEW

File: lib/l10n/app_en.arb [hunk 1]
```diff
@@ -17,5 +17,369 @@
...
+    "@entryphonenumber": {
+        "description": "Please enter the phone number!"
+    },
+
+    "myphysicians": "My Physicians",
+    "@myphysicians": {
+        "description": "My Physicians"
+    },
+
+    "practice": "Practice",
```
Comment (S1rDan) [inline lib/l10n/app_en.arb:296]: Is this the correct term in the context? The german translations "Praxis" could mean multiple things.
Maybe "surgery" would fit better.

File: lib/l10n/app_en.arb [hunk 2]
```diff
@@ -17,5 +17,369 @@
...
+    "@entrypracticename": {
+        "description": "Please enter the practice name!"
+    },
+
+    "mymedicaldata": "My medical data",
+    "@medicaldata": {
+        "description": "My medical data"
+    },
+
+    "achievedvalues": "Today's achieved values",
```
Comment (S1rDan) [inline lib/l10n/app_en.arb:366]:
```suggestion
    "contributedValues": "Today's contributed values",
```

Other option could be "achievedEntries"

Other suggestions?

**Review by S1rDan — CHANGES_REQUESTED** [review]
Review of arb files.

File: lib/screens/overview_screen/widgets/graph_section.dart [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+*  along with this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import 'package:fl_chart/fl_chart.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+/// Widget which shows a graph
+class GraphSection extends StatefulWidget {
+  // ignore: public_member_api_docs
```
Comment (S1rDan) [inline lib/screens/overview_screen/widgets/graph_section.dart:25]:
```suggestion
  /// GraphSection constructor
```

Using just a simple comment resolves the issue and is no more effort than the ignore statement.

File: lib/screens/overview_screen/widgets/my_health_section.dart [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:picos/screens/my_medications_screen/my_medications_screen.dart';
+import 'package:picos/screens/physicians_screen.dart';
+import 'package:picos/screens/family_members_screen.dart';
+
+/// Widget which displays health-related information
+class MyHealthSection extends StatelessWidget {
+  // ignore: public_member_api_docs
```
Comment (S1rDan) [inline lib/screens/overview_screen/widgets/my_health_section.dart:27]: Using just a simple comment resolves the issue and is no more effort than the ignore statement.

File: lib/screens/overview_screen/widgets/my_health_section.dart [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+  const MyHealthSection({Key? key}) : super(key: key);
+
+  // final double h = MediaQuery.of(context).size.height;
```
Comment (S1rDan) [inline lib/screens/overview_screen/widgets/my_health_section.dart:30]: Is this commented out code still relevant?
Can it be removed or is it possible to use double.infinity for size values?

File: lib/screens/overview_screen/widgets/my_health_section.dart [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+            ),
+          ),
+          // This is a horizontal line
+          // be my guest to make it better if you know how
+          SizedBox(
+            height: 1,
+            child: Container(
+              color: Colors.white,
+            ),
+          ),
```
Comment (S1rDan) [inline lib/screens/overview_screen/widgets/my_health_section.dart:53]:
```suggestion
          const Divider(
            height: 1,
            color: Colors.white,
            thickness: 1,
          ),
```

**Review by S1rDan — CHANGES_REQUESTED** [review]
Reviewing main_screen and overwiew_screen.

File: lib/l10n/app_en.arb [hunk 7]
```diff
@@ -17,5 +17,369 @@
...
+        "description": "Please enter a valid website!"
+    },
+
+    "entrypracticename": "Please enter the practice name!",
+    "@entrypracticename": {
+        "description": "Please enter the practice name!"
+    },
+
+    "mymedicaldata": "My medical data",
+    "@medicaldata": {
```
[the remark this answers is not in the document]
Comment (mustiwebp) [reply lib/l10n/app_en.arb:362]: Yes, of course! To be comitted!

File: lib/l10n/app_en.arb [hunk 8]
```diff
@@ -17,5 +17,369 @@
...
+    "@abort": {
+        "description": "Abort"
+    },
+
+    "proceed": "Continue",
+    "@proceed": {
+        "description": "Continue"
+    },
+
+    "myentries": "My Entries",
```
[the remark this answers is not in the document]
Comment (mustiwebp) [reply lib/l10n/app_en.arb:236]: I now applied camelCase to all denotations. To be comitted!

File: lib/l10n/app_en.arb [hunk 9]
```diff
@@ -17,5 +17,369 @@
...
+    "@more": {
+        "description": "More"
+    },
+
+    "notifications": "Motifications",
+    "@notifications": {
+        "description": "Notifications"
+    },
+
+    "profil": "Profil",
```
[the remark this answers is not in the document]
Comment (mustiwebp) [reply lib/l10n/app_en.arb:131]: Corrected! To be comitted!

File: lib/l10n/app_en.arb [hunk 10]
```diff
@@ -17,5 +17,369 @@
...
+    "@delete": {
+        "description": "Delete"
+    },
+
+    "mister": "Mr.",
+    "@mister": {
+        "description": "Mr."
+    },
+
+    "mistress": "Mrs.",
```
[the remark this answers is not in the document]
Comment (mustiwebp) [reply lib/l10n/app_en.arb:186]: Yeah, of course! It has been changed. To be comitted!

File: lib/l10n/app_en.arb [hunk 11]
```diff
@@ -17,5 +17,369 @@
...
+    "@profil": {
+        "description": "Profil"
+    },
+
+    "privacynotice": "Privacy Notice",
+    "@privacynotice": {
+        "description": "Privacy Notice"
+    },
+
+    "imprint": "Imprint",
```
[the remark this answers is not in the document]
Comment (mustiwebp) [reply lib/l10n/app_en.arb:141]: Done! To be comitted!

File: lib/l10n/app_en.arb [hunk 12]
```diff
@@ -17,5 +17,369 @@
...
+    "@myentries": {
+        "description": "My Entries"
+    },
+
+    "startentry": "Start Entry",
+    "@startentry": {
+        "description": "Start Entry"
+    },
+
+    "submitdata": "The entered will be submitted.",
```
[the remark this answers is not in the document]
Comment (mustiwebp) [reply lib/l10n/app_en.arb:246]: Fair enough. Changed! To be comitted!

File: lib/l10n/app_en.arb [hunk 13]
```diff
@@ -17,5 +17,369 @@
...
+    "@wife": {
+        "description": "Wife"
+    },
+
+    "entryemail": "Please enter the e-mail-address!",
+    "@entryemail": {
+        "description": "Please enter the e-mail-address!"
+    },
+
+    "entryvalidemail": "Please enter a valid e-mail-address!",
```
[the remark this answers is not in the document]
Comment (mustiwebp) [reply lib/l10n/app_en.arb:266]: The German transmission was just missing. Sorry for that. To be comitted!

File: lib/l10n/app_en.arb [hunk 14]
```diff
@@ -17,5 +17,369 @@
...
+    },
+
+    "entryfamilyname": "Please enter the family name!",
+    "@entryfamilyname": {
+        "description": "Please enter the family name"
+    },
+
+    "entryaddress": "Please enter the address!",
+    "@entryaddress": {
+        "description": "Please enter the family name"
```
[the remark this answers is not in the document]
Comment (mustiwebp) [reply lib/l10n/app_en.arb:283]: Alright, changed! To be comitted!

File: lib/l10n/app_en.arb (hunk 1 above)
Comment (mustiwebp) [reply lib/l10n/app_en.arb:296]: Well, as far as I know "surgery" is just a room where a physician does his job. In my opinion, "practice" would be more appropriate. I let it open for discussion :)

File: lib/screens/overview_screen/widgets/graph_section.dart [hunk 15]
```diff
@@ -0,0 +1,270 @@
...
+              const TextStyle(
+                color: Colors.white,
+                fontWeight: FontWeight.bold,
+              ),
+            );
+          },
+        ),
+      );
+
+  // Widget getTitles(double value, TitleMeta meta) {
```
[the remark this answers is not in the document]
Comment (mustiwebp) [reply lib/screens/overview_screen/widgets/graph_section.dart:139]: In general, a bunch of code that is commented out is not a comment in this sprit. I don't know yet whether it'll be used in future, but as far as I can see it works without the commented code :)

File: lib/screens/overview_screen/widgets/progress_section.dart [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+/// Widget which is used for displaying
+/// the progress bar in the corresponding section on the "overview"-screen
+class ProgressSection extends StatelessWidget {
+  // ignore: public_member_api_docs
+  const ProgressSection({Key? key}) : super(key: key);
+
+  /// Declarion of the value (percentage) which is shown on the progress bar
```
[the remark this answers is not in the document]
Comment (mustiwebp) [reply lib/screens/overview_screen/widgets/progress_section.dart:28]: Yeah, an unnecessary typo :) I changed it, of course. To be comitted!

File: lib/screens/overview_screen/overview_screen.dart [hunk 17]
```diff
@@ -0,0 +1,34 @@
+import 'package:flutter/material.dart';
+import 'package:picos/screens/overview_screen/widgets/graph_section.dart';
+import 'package:picos/screens/overview_screen/widgets/my_health_section.dart';
+import 'widgets/input_card_section.dart';
+import 'widgets/progress_section.dart';
+
+/// Main widget for the
```
[the remark this answers is not in the document]
Comment (mustiwebp) [reply lib/screens/overview_screen/overview_screen.dart:7]: Alright! Changed! To be comitted!

**Comment by mustiwebp** [comment]
I noticed that in some files (add_medication_screen/add_medication_screen_label.dart, add_medication_screen/add_medication_screen.dart, my_medications_screen/medication_card_tile.dart, my_medications_screen/medication_card.dart, my_medications_screen/medication_list.dart, my_medications_screen/my_medications_screen.dart) the GNU licence text is missing. 

Isn't it mandatory in our project to put this licence text at the top of each file?

File: lib/l10n/app_en.arb (hunk 1 above)
Comment (S1rDan) [reply lib/l10n/app_en.arb:296]: Afaik "practice" can be used, but it is ambiguous.
But we could leave it as it is and change it later if someone complains about it.

File: lib/screens/overview_screen/widgets/graph_section.dart (hunk 15 above)
[the remark this answers is not in the document]
Comment (S1rDan) [reply lib/screens/overview_screen/widgets/graph_section.dart:139]: As a rule of thumb unused code shouldn't be commited unless it has a justification relevant for the whole project.
If this is just a code segment that you keep for a potential later use, you could use the git shelve feature for that or just store it by other means.

**Comment by S1rDan** [comment]
> 

Good question. I would also like to know if it's really necessary to insert this in each file throughout the project.
For me it looks just like duplication, but maybe it is legally mandatory. Anyone who can tell?

How do you handle the inclusions? Is there a way to make the IDE automatically include this paragraph on file creation or do you just paste it manually?

File: lib/screens/family_members_add_screen.dart [hunk 18]
```diff
@@ -0,0 +1,332 @@
...
+*
+*    You should have received a copy of the GNU General Public License
+*    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+/// contains the gender of the corresponding family member
+// ignore: public_member_api_docs
```
Comment (S1rDan) [inline lib/screens/family_members_add_screen.dart:22]: Another instance of an unnecessary ignore

File: lib/screens/family_members_add_screen.dart [hunk 19]
```diff
@@ -0,0 +1,332 @@
...
+  if (key.currentState!.validate()) {
+    ScaffoldMessenger.of(context).showSnackBar(
+      SnackBar(
+        content: Text(
+          AppLocalizations.of(context)!.submitData,
+        ),
+      ),
+    );
+  }
+}
```
Comment (S1rDan) [inline lib/screens/family_members_add_screen.dart:332]: Is this code supposed to execute when pressing the "add" Button in the family members screen?
I'm getting an error when doing this. Is this an accounted behaviour?

File: lib/screens/physicians_add_screen.dart [hunk 20]
```diff
@@ -0,0 +1,363 @@
...
+*
+*    You should have received a copy of the GNU General Public License
+*    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+/// contains the gender of the corresponding physician
+// ignore: public_member_api_docs
```
Comment (S1rDan) [inline lib/screens/physicians_add_screen.dart:22]: Unecessary ignore statement.

**Review by S1rDan — CHANGES_REQUESTED** [review]
Reviewing questionaire, family, physicians and menu screens.

File: lib/l10n/app_de.arb [hunk 21]
```diff
@@ -2,5 +2,79 @@
...
+    "brother": "Bruder",
+    "sister": "Schwester",
+    "edit": "Bearbeiten",
+    "delete": "Löschen",
+    "mister": "Herr",
+    "mistress": "Frau",
+    "choosefamilymember": "Wählen Sie Ihren Familienangehörigen aus.",
+    "title": "Anrede",
+    "firstname": "Vorname",
+    "familyname": "Nachname",
```
[the remark this answers is not in the document]
Comment (mustiwebp) [reply lib/l10n/app_de.arb:44]: Of course!

File: lib/screens/family_members_add_screen.dart (hunk 19 above)
Comment (mustiwebp) [reply lib/screens/family_members_add_screen.dart:332]: What kind of error do you get? Actually, it shouldn't do anything after pressing the "add" Button

File: lib/screens/physicians_add_screen.dart (hunk 20 above)
Comment (mustiwebp) [reply lib/screens/physicians_add_screen.dart:22]: Done!

File: lib/screens/family_members_add_screen.dart (hunk 19 above)
Comment (S1rDan) [reply lib/screens/family_members_add_screen.dart]: There is an onPressed function that uses a `Navigator.push()` to FamilyMembersAdd().
Doing this I'm getting a red screen with an assertion error.

Comment (mustiwebp) [reply lib/screens/family_members_add_screen.dart]: This problem should be fixed now!

File: lib/l10n/app_de.arb (hunk 21 above)
[the remark this answers is not in the document]
Comment (S1rDan) [reply lib/l10n/app_de.arb]: In the german file there is still "profil" instead of "profile" for the key.

**Comment by S1rDan** [comment]
I've added one additional comment.
Apart from that, @inchw0rm as a contributor and maintainer needs to take part in the discussion.

File: lib/screens/family_members_add_screen.dart (hunk 19 above)
Comment (inchw0rm) [reply lib/screens/family_members_add_screen.dart]: > I noticed that in some files (add_medication_screen/add_medication_screen_label.dart, add_medication_screen/add_medication_screen.dart, my_medications_screen/medication_card_tile.dart, my_medications_screen/medication_card.dart, my_medications_screen/medication_list.dart, my_medications_screen/my_medications_screen.dart) the GNU licence text is missing.
> 
> Isn't it mandatory in our project to put this licence text at the top of each file?

@S1rDan 

The GPL 3.0 has got a "how to apply" section at the bottom and it says the following:

> How to Apply These Terms to Your New Programs
> If you develop a new program, and you want it to be of the greatest possible use to the > public, the best way to achieve this is to make it free software which everyone can
> redistribute and change under these terms.

> To do so, attach the following notices to the program. It is safest to attach them to
> the start of each source file to most effectively state the exclusion of warranty; and
> each file should have at least the “copyright” line and a pointer to where the full
> notice is found.

**Comment by inchw0rm** [comment]
> I noticed that in some files (add_medication_screen/add_medication_screen_label.dart, add_medication_screen/add_medication_screen.dart, my_medications_screen/medication_card_tile.dart, my_medications_screen/medication_card.dart, my_medications_screen/medication_list.dart, my_medications_screen/my_medications_screen.dart) the GNU licence text is missing.
> 
> Isn't it mandatory in our project to put this licence text at the top of each file?

@S1rDan 

The GPL 3.0 has got a "how to apply" section at the bottom and it says the following:

> How to Apply These Terms to Your New Programs
> If you develop a new program, and you want it to be of the greatest possible use to the > public, the best way to achieve this is to make it free software which everyone can
> redistribute and change under these terms.

> To do so, attach the following notices to the program. It is safest to attach them to
> the start of each source file to most effectively state the exclusion of warranty; and
> each file should have at least the “copyright” line and a pointer to where the full
> notice is found.

**Comment by inchw0rm** [comment]
@mustiwebp this is a very large PR, I'd appreciated if you could split it in several smaller ones.

**Comment by S1rDan** [comment]
> > I noticed that in some files (add_medication_screen/add_medication_screen_label.dart, add_medication_screen/add_medication_screen.dart, my_medications_screen/medication_card_tile.dart, my_medications_screen/medication_card.dart, my_medications_screen/medication_list.dart, my_medications_screen/my_medications_screen.dart) the GNU licence text is missing.
> > Isn't it mandatory in our project to put this licence text at the top of each file?
> 
> @S1rDan
> 
> The GPL 3.0 has got a "how to apply" section at the bottom and it says the following:
> 
> > How to Apply These Terms to Your New Programs
> > If you develop a new program, and you want it to be of the greatest possible use to the > public, the best way to achieve this is to make it free software which everyone can
> > redistribute and change under these terms.
> 
> > To do so, attach the following notices to the program. It is safest to attach them to
> > the start of each source file to most effectively state the exclusion of warranty; and
> > each file should have at least the “copyright” line and a pointer to where the full
> > notice is found.

Good to know, thanks!

File: lib/l10n/app_de.arb (hunk 21 above)
[the remark this answers is not in the document]
Comment (mustiwebp) [reply lib/l10n/app_de.arb]: Yes, done!

**Comment by inchw0rm** [comment]
Sorry it takes so long, I was a little busy, I am still in the process of reviewing the changes. Please give me some more time.

**Comment by S1rDan** [comment]
> I also have something to say about that: I asked you if the merge is only for the presentation, since you and @S1rDan merged your repos, you assured me that the merge was for the presentation only, but here we are, now we've got a PR with code you have claimed is only for the demonstration, and furthermore contains the code I explicitly asked to fix and asked questions about that remain unanswered.

What is the problem you are talking about here? I don't remember that. Can you please show me what you mean?

**Comment by S1rDan** [comment]
> Sorry it takes so long, I was a little busy, I am still in the process of reviewing the changes. Please give me some more time.

Is there further progress in this regard?

**Comment by inchw0rm** [comment]
> > Sorry it takes so long, I was a little busy, I am still in the process of reviewing the changes. Please give me some more time.
> 
> Is there further progress in this regard?

No, the PR is so large even github fails to load it in a reasonable amount of time and this is not the only project I am working on.

**Comment by S1rDan** [comment]
> > > Sorry it takes so long, I was a little busy, I am still in the process of reviewing the changes. Please give me some more time.
> > 
> > 
> > Is there further progress in this regard?
> 
> No, the PR is so large even github fails to load it in a reasonable amount of time and this is not the only project I am working on.

What process is taking so long loading? Opening this page or the changed files page? Or is it a process on the local machine?

File: lib/screens/add_medication_screen/add_medication_screen.dart [hunk 22]
```diff
@@ -0,0 +1,293 @@
...
+                        SaveMedication(
+                          Medication(
+                            compound: _compound!,
+                            morning: _morning,
+                            noon: _noon,
+                            evening: _evening,
+                            night: _night,
+                          ),
+                        ),
+                      );
```
Comment (mustiwebp) [inline lib/screens/add_medication_screen/add_medication_screen.dart:170]: Bracket formattion of "add" might be inappropriate

**Review by mustiwebp — COMMENTED** [review]
I found some formatting issues. Have a look!

**Comment by S1rDan** [comment]
> I found some formatting issues. Have a look!

These is essentially the very same issue in each single instance.
I don't know any solution to this except to fix it manually, since _flutter format_ don't work for me here.

I think the code isn't unreadable at this point, but if it bothers you, please suggest a better solution.

**Comment by mustiwebp** [comment]
> > I found some formatting issues. Have a look!
> 
> These is essentially the very same issue in each single instance. I don't know any solution to this except to fix it manually, since _flutter format_ don't work for me here.

Okay

**Pull request merged**